core/src/main/java/org/apache/calcite/rel/core/Aggregate.java

```diff
@@ -199,9 +227,9 @@ private boolean isPredicate(RelNode input, int index) {
    * Creates an Aggregate by parsing serialized output.
    */
   protected Aggregate(RelInput input) {
-    this(input.getCluster(), input.getTraitSet(), input.getInput(),
-        input.getBitSet("group"), input.getBitSetList("groups"),
-        input.getAggregateCalls("aggs"));
+    this(input.getCluster(), input.getTraitSet(), new ArrayList<>(),
```

Comment:
nit: As the internal data structure is Immutable, you can use `Collections.emptyList()` to create an ImmutableList directly that  using `new ArrayList<>()`.

Reply:
Thanks for the review @amaliujia , i missed you comments when i commit the code, sorry for that.

Reply:
no worries. It's just a `nit`. Thanks for this contribution and quickly review. I can build hot key hint on top of it now.